src/modules/IscsiClientLib.rb

```diff
@@ -748,44 +763,46 @@ def deleteRecord
       ret
     end
 
-    # get (manual/onboot) status of target connecting
+    # Get info about current iSCSI node
+    #
+    # return [Hash]    stdout of 'iscsiadm -m node -I <iface> -T <target> -p <ip>'
+    #                  converted to a hash
+    def getCurrentNodeValues
+      node_val = {}
+
+      ret = SCR.Execute(path(".target.bash_output"),
+                        GetAdmCmd("-m node -I #{@currentRecord[2]||"default"} -T #{@currentRecord[1]||""} -p #{@currentRecord[0]||""}"))
+      return node_val unless (ret["exit"] || -1) == 0
+
+      node_val = nodeInfoToMap(ret["stdout"] || "")
+    end
+
+    # Get (manual/onboot/automatic) status of target connection
+    #
+    # return [String]   startup status of the iSCSI node
+    #
     def getStartupStatus
       status = ""
-      Builtins.y2milestone("Getting status of record %1", @currentRecord)
-      retcode = Convert.convert(
-        SCR.Execute(
-          path(".target.bash_output"),
-          GetAdmCmd(
-            Builtins.sformat(
-              "-m node -I %3 -T %1 -p %2",
-              Ops.get(@currentRecord, 1, ""),
-              Ops.get(@currentRecord, 0, ""),
-              Ops.get(@currentRecord, 2, "default")
-            )
-          )
-        ),
-        :from => "any",
-        :to   => "map <string, any>"
-      )
-      if Ops.greater_than(
-          Builtins.size(Ops.get_string(retcode, "stderr", "")),
-          0
-        )
-        return ""
-      end
-      Builtins.foreach(
-        Builtins.splitstring(Ops.get_string(retcode, "stdout", ""), "\n")
-      ) do |row|
-        if Builtins.issubstring(row, "node.conn[0].startup")
-          status = Ops.get(Builtins.splitstring(row, " "), 2, "")
-          raise Break
+      log.info "Getting status of record #{@currentRecord}"
+
+      curr_node = getCurrentNodeValues
+      ibft = getiBFT
+
+      if (!ibft.empty?)
+        # check whether current session/record is iBFT device 
+        if (ibft["iface.transport_name"] == curr_node["iface.transport_name"]) &&
+           (ibft["iface.initiatorname"] == curr_node["iface.initiatorname"]) &&
+           (ibft["node.name"] == curr_node["node.name"]) &&
+           (ibft["node.conn[0].address"] == curr_node["node.conn[0].address"])
+          # always show status "onboot" (startup value from node doesn't matter for iBFT) 
+          status = "onboot"
+          log.info "Startup status for iBFT is always onboot"
```

Comment:
It's probably not correct (I also came to this conclusion overnight).
I will recheck.